run-build-functions.sh

```diff
@@ -779,6 +820,54 @@ restore_cwd_cache() {
   move_cache "$NETLIFY_CACHE_DIR/$1" "$PWD/$1" "restoring cached $2"
 }
 
+#
+# Restores node_modules dirs cached for js workspaces
+# See https://github.com/netlify/pod-workflow/issues/139/ for more context
+#
+# Expects:
+# $@ each argument should be a package location relative to the repo's root
+restore_js_workspaces_cache() {
+  local locations=("$@")
+  local cache_dir="$NETLIFY_CACHE_DIR/js-workspaces"
+
+  # Retrieve each workspace node_modules
+  for location in "${locations[@]}"; do
+    move_cache "$cache_dir/$location/node_modules" "$NETLIFY_REPO_DIR/$location/node_modules" "restoring workspace $location node modules"
+  done
+  # Retrieve hoisted node_modules
+  move_cache "$cache_dir/node_modules" "$NETLIFY_REPO_DIR/node_modules" "restoring workspace root node modules"
+  # Keep a record of the workspaces in the project in order to cache them later
+  echo "${locations[@]}" > "$NETLIFY_TMP_DIR/.workspace_locations"
```

Comment:
Because we need to have the ability to cache each workspace location at the end of the run we use the `restore_js_workspaces_cache` function as a way to set a temporary file, containing the package locations, that we can later use when caching directories.

Reply:
Would a variable work instead of a file?
This would remove the need for `NETLIFY_TMP_DIR` and would be faster.

Reply:
@ehmicky we could use a global variable, this will work as long the `buildbot` runs these functions as part of the same shell execution I believe (which is the case now). It would however create a tighter coupling with the way we execute the `run-build.sh` script, meaning if we ever plan to install dependencies on a separate execution of which we cache things this would not work. That was my reasoning for using a file instead, but if you feel like it's too much happy to move to a different strategy.

Reply:
Oh, I see what you mean.
I am wondering whether we will be in a situation where those different functions would end up in different processes.
Since this is not sure, should we defer using IPC (through a file or other mean), and use a global variable instead?

Reply:
Yep 👍 I guess we can cross that bridge when it's time to. Addressed via https://github.com/netlify/build-image/commit/70c8a36e04ac440911ec776386d80536063768c6 @ehmicky